tools/hidl/hidlcomp.cpp

```diff
@@ -6608,28 +6602,20 @@ void EspServInfo::write_esp_client()
         outs("\tif(state!=NULL)\n");
         outs("\t\tstate->Link();\n\n");
 
-        
-        outs("#ifdef _WIN32\n");
-        outs("\t_beginthread(espWorkerThread, 0, (void *)(IRpcRequestBinding *)(esprequest));\n");
-        outs("#else\n");
-
-        outs("\tpthread_attr_t attr;\n");
-        outs("\tpthread_attr_init(&attr);\n");
-        outs("\tpthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM);\n");
-        outs("\tpthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);\n");
-        outs("\tpthread_attr_setstacksize(&attr, 0x10000);\n");
-        outs("\tThreadId threadid;\n");
-        outs("\tint status;\n");
-        outs("\tdo\n");
+        outs("\n");
+        outf("\tclass CClient%sThread: public Thread\n", name_);
```

Comment:
Some of this code could be moved to a base class:

	class CServiceThread: public Thread
	{
	public:
		Owned<IRpcRequestBinding> m_esprequest;
		CClientWsLogServiceThread(IRpcRequestBinding *_esprequest) 
                { m_esprequest.setown(_esprequest); setStackSize(0x10000); }
	}

which would reduce the generated code.

Reply:
I would go further and add a single helper function:

`    executeServiceThread(CClient%s::espWorkerThread)
`
and move all of this code out of the generated file.  Is a long term goal to switch this all to dynamic esdl?